coffea/processor/executor.py

```diff
@@ -107,6 +107,53 @@ def futures_executor(items, function, accumulator, workers=1, status=True, unit=
     return accumulator
 
 
+def _dask_reduce(items):
+    if len(items) == 0:
+        raise ValueError("Empty list to reduction?!  Should not happen")
+    elif len(items) == 1:
+        return items[0]
+    else:
+        out = items[0]
+        for item in items[1:]:
+            out += item
+        return out
+
+
+def dask_executor(items, function, accumulator, **kwargs):
+    """Execute using dask futures
+
+    Parameters
+    ----------
+        items : list
+            List of input arguments
+        function : function
+            A function to be called on each input, which returns an accumulator instance
+        accumulator : AccumulatorABC
+            An accumulator to collect the output of the function
+        client : distributed.client.Client
+            A dask distributed client instance
+        treereduction : int, optional
+            Tree reduction factor for output accumulators (default: 20)
+        status : bool, optional
+            If true (default), enable progress bar
+    """
+    client = kwargs.pop('client')
+    ntree = kwargs.pop('treereduction', 20)
+    status = kwargs.pop('status', True)
+    futures = client.map(function, items, **kwargs)
```

Comment:
For large tasks, the graph construction hangs for an annoyingly long time.  I am not sure if I'm setting this up improperly, or what.

Reply:
`item` includes the processor instance, and since the rest of the tuple changes, dask doesn't see that its always the same.  This leads to shipping the processor `N` times for n tasks.  Solution should be to create a closure function carrying the processor so that `item` is lightweight.